src/main/java/jenkins/plugins/openstack/compute/JCloudsSlaveTemplate.java

```diff
@@ -279,12 +277,20 @@ public boolean canProvision(final Label label) {
     }
 
     /*package for testing*/ @CheckForNull String getUserData() {
-        Config userData = ConfigProvider.all().get(UserDataConfig.UserDataConfigProvider.class).getConfigById(getEffectiveSlaveOptions().getUserDataId());
 
-        return (userData == null || userData.content.isEmpty())
-                ? null
-                : userData.content
-        ;
+        Executor executor = Executor.currentExecutor();
+        if (executor != null) {
+            Queue.Executable currentExecutable = executor.getCurrentExecutable();
+            if (currentExecutable != null) {
+                Config userData = ConfigFiles.getByIdOrNull((Run<?, ?>) currentExecutable, getEffectiveSlaveOptions().getUserDataId());
```

Comment:
The tests should  be able to verify we send the text content to mocked openstack as "user data". It would be great to have test that demonstrates the folder support.